tests/topotests/multicast-pim-sm-topo1/test_multicast_pim_sm_topo1.py

```diff
@@ -0,0 +1,1697 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND VMWARE DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL VMWARE BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+Following tests are covered to test multicast pim sm:
+
+Test steps
+- Create topology (setup module)
+- Bring up topology
+
+Following tests are covered:
+1. TC_1_1: Verify Multicast data traffic with static RP, (*,g) and
+   (s,g) OIL updated correctly
+2. TC_1_2: Verify Multicast data traffic with static RP, (*,g) and
+   (s,g) OIL updated correctly
+3. TC_4: Verify removing the RP should not impact the multicast
+   data traffic
+4. TC_5: Verify (*,G) and (S,G) entry populated again after clear the
+   PIM nbr and mroute from FRR node
+5. TC_9: Verify (s,g) timeout from FHR and RP when same receive
+   exist in LHR , FHR and RP
+6. TC_19: Verify mroute detail when same receiver joining 5
+    different sources
+7. TC_16: Verify (*,G) and (S,G) populated correctly
+    when FRR is the transit router
+8. TC_23: Verify (S,G) should not create if RP is not reachable
+9. TC_24: Verify modification of IGMP query timer should get update
+    accordingly
+10. TC_25: Verify modification of IGMP max query response timer
+    should get update accordingly
+"""
+
+import os
+import sys
+import json
+import time
+import datetime
+from time import sleep
+import pytest
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# Required to instantiate the topology builder class.
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+from mininet.topo import Topo
+
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    write_test_footer,
+    step,
+    iperfSendIGMPJoin,
+    addKernelRoute,
+    reset_config_on_routers,
+    iperfSendTraffic,
+    kill_iperf,
+    shutdown_bringup_interface,
+    kill_router_daemons,
+    start_router,
+    start_router_daemons,
+    stop_router,
+    required_linux_kernel_version,
+    topo_daemons,
+)
+from lib.pim import (
+    create_pim_config,
+    create_igmp_config,
+    verify_igmp_groups,
+    verify_ip_mroutes,
+    verify_pim_interface_traffic,
+    verify_upstream_iif,
+    verify_pim_neighbors,
+    verify_pim_state,
+    verify_ip_pim_join,
+    clear_ip_mroute,
+    clear_ip_pim_interface_traffic,
+    verify_igmp_config,
+    clear_ip_mroute_verify
+)
+from lib.topolog import logger
+from lib.topojson import build_topo_from_json, build_config_from_json
+
```

Comment:
Please add pytest module level markers as described in the dev guide: http://docs.frrouting.org/projects/dev-guide/en/latest/topotests.html#markers

Reply:
Done.